andela/travel_tool_back
# #160710730  Mark room as faulty

#### What does this PR do?
It adds the functionality to update the faulty property of a room to either false or true. This marks a room for maintenance.

#### Description of Task to be completed?
- Add an endpoint `/api/v1/room/:id`
- Add a controller to map the endpoint to and edit fault property of a room.
- Add a validator to ensure data in the body is either `true` or `false`

#### How should this be manually tested?
- Clone the repository and enter the project's folder
- Checkout to `ft-mark-room-as-faulty-#160710730`
- Assign yourself the travel admin role by changing your roleId to 29187 in the users table
- Start the application using 'make start'
- Open postman and provide your token in the authorization header.
- If you do not have a guest house id, see [this](https://github.com/andela/travel_tool_back/pull/65) for guesthouse structure to create one
- Make a PUT request to `/api/v1/room/:id`. Ensure to add the `id` as the ID of any room.
- user {fault:true} or {fault:false} as the PUT request body.

RUNNING TESTS
- `make test`

#### Any background context you want to provide?
- Ensure you have logged in the front-end and you are saved as a travel admin.
- Ensure you create a guesthouse
#### What are the relevant pivotal tracker stories?
[#160710730](https://www.pivotaltracker.com/story/show/160710730)
#### Screenshots (if appropriate)
MARKING A ROOM AS FAULTY
<img width="1238" alt="screen shot 2018-10-04 at 09 53 12" src="https://user-images.githubusercontent.com/23402614/46457537-5e515380-c7bb-11e8-84fc-fa0fc11ece0b.png">

MARK ROOM THAT DOES NOT EXIST
<img width="1232" alt="screen shot 2018-10-04 at 09 53 22" src="https://user-images.githubusercontent.com/23402614/46457551-68735200-c7bb-11e8-9e54-5f24dcd7d836.png">

#### Questions:
